# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/promise/Promise.cpp
# Commit: 1c7a70ea55e8
# Full Hash: 1c7a70ea55e881a42d6bbe2375945f8b532edbb5
# Author: Kris Maglione <maglione.k@gmail.com>
# Date: 2020-02-26 09:27:57
# Regressor Bug: 1535617
# File Overlap Count: 1
# Description:
#   Bug 1535617: Part 2 - Clear weak references for most cycle collected objects on unlink. r=mccr8
#   
#   This covers most cycle collected objects which support weak references, but
#   not the ones which inherit from a cycle collected class and don't do any cycle
#   collection on their own.
# ==============================================================================

diff -r 275198c0a966 -r 1c7a70ea55e8 dom/promise/Promise.cpp
--- a/dom/promise/Promise.cpp	Tue Feb 25 19:44:27 2020 +0000
+++ b/dom/promise/Promise.cpp	Tue Feb 25 19:44:39 2020 +0000
@@ -53,6 +53,7 @@
 
 NS_IMPL_CYCLE_COLLECTION_UNLINK_BEGIN(Promise)
   NS_IMPL_CYCLE_COLLECTION_UNLINK(mGlobal)
+  NS_IMPL_CYCLE_COLLECTION_UNLINK_WEAK_PTR
   tmp->mPromiseObj = nullptr;
 NS_IMPL_CYCLE_COLLECTION_UNLINK_END
 